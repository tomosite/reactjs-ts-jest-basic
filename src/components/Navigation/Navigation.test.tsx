import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Navigation } from '.';
import { LinkItemProps } from '.';
import renderer from 'react-test-renderer';

describe('Navigation component', () => {
  let mockFn: () => void;
  let linkItems: Array<LinkItemProps>;

  beforeEach(() => {
    mockFn = jest.fn();
    linkItems = [
      { id: '1', children: <>item dummy 1</>, onClick: mockFn },
      { id: '2', children: <>item dummy 2</>, onClick: mockFn }
    ];
  });

  test('snapshot navigation', () => {
    const domTree = renderer.create(<Navigation links={linkItems} />);
    expect(domTree).toMatchSnapshot();
  });

  test('click the navigation link item', () => {
    render(<Navigation links={linkItems} />);
    const link_elem = screen.getByText(/item dummy 1/);

    fireEvent.click(link_elem, { button: 0 });
    expect(mockFn).toHaveBeenCalled();
  });
});
