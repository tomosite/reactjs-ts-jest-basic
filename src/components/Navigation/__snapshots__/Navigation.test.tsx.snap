// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component snapshot navigation 1`] = `
<nav>
  <ul>
    <li
      id="1"
      onClick={[Function]}
    >
      item dummy 1
    </li>
    <li
      id="2"
      onClick={[Function]}
    >
      item dummy 2
    </li>
  </ul>
</nav>
`;
